Packs/DigitalShadows/Integrations/ReliaQuestGreyMatterDRPIncidents/ReliaQuestGreyMatterDRPIncidents.py

```diff
@@ -0,0 +1,964 @@
+            # based on event-action="create" and status="rejected" on the triage item event
+            auto_closed = data_item[EVENT][EVENT_ACTION] == EVENT_ACTION_CREATE and data_item[TRIAGE_ITEM][
+                STATE] == NON_INGESTIBLE_TRIAGE_ITEM_STATES
+            data_item[AUTO_CLOSED] = auto_closed
+            data_item = removing_unwanted_data(data_item)
+            data.append(data_item)
+        return data
+
+    def stringyfyAlert(self, alert):
+        if alert.get('risk-factors'):
+            alert.update({'risk-factors': json.dumps(alert.get('risk-factors'))})
+        if alert.get('mitre-attack-mapping'):
+            alert.update({'mitre-attack-mapping': json.dumps(alert.get('mitre-attack-mapping'))})
+        if alert.get('validation'):
+            alert.update({'validation': json.dumps(alert.get('validation'))})
+        return alert
+
+    def poll_triage(self, event_created_after, event_num_start=0, limit=100, alert_risk_types=[RISK_TYPE_ALL],
+                    risk_level=[RISK_LEVEL_ALL], should_ingest_closed=True):
+        """
+        A single poll of the triage API for new events, fully populating any new events found.
+
+        Calls a provided callback method with the fully-populated data.
+
+        Returns the largest event-num from the triage item events that were processed.
+        """
+        LOG(
+            "Polling triage items. Event num start: {}, Event created after: {}, Limit: {} risk_level: {} "
+            "alert_risk_types: {}".format(
+                event_num_start,
+                event_created_after,
+                limit, risk_level, alert_risk_types))
+        risk_types_filter = []
+
+        if RISK_TYPE_ALL not in alert_risk_types and len(alert_risk_types) > 0:
+            risk_types_filter = alert_risk_types
+
+        events = get_triage_item_events(self.request_handler, event_created_after=event_created_after,
+                                        risk_types=risk_types_filter, event_num_after=event_num_start, limit=limit)
+        if events:
+            max_event_num = max([e['event-num'] for e in events])
+        # Only ingesting events with create action event
+        if events:
+            events = [event for event in events if event[EVENT_ACTION].lower() == EVENT_ACTION_CREATE]
+
+        risk_level_filter = []
+
+        if RISK_LEVEL_ALL not in risk_level and len(risk_level) > 0:
+            risk_level_filter = risk_level
+
+        # filtering events by risk level
+        if risk_level_filter:
+            events = [event for event in events if event[RISK_LEVEL] in risk_level_filter]
+
+        if not events:
+            LOG("No events were fetched. Event num start: {}, Event created after: {}, Limit:"
+                " {}, risk_level: {}, alert_risk_types: {}".format(
+                    event_num_start, event_created_after, limit, risk_level, alert_risk_types))
+            return RQPollResult(max_event_num, [])
```

Comment:
When we don't receive events it will fall, because the `max_event_num` variable has not been defined

am i wrong?

Reply:
Yes you are right pushing change to handle that.